src/main/java/trackitnus/logic/Logic.java

```diff
@@ -55,6 +57,38 @@
      */
     ObservableList<Lesson> getFilteredLessonList();
 
+    /**
+     * @param date The date to query
+     * @return all lessons happens on that date
+     */
+    ObservableList<Lesson> getDayUpcomingLessons(LocalDate date);
+
+    /**
+     * @param code The module code to query
+     * @return the list of lesson for a specific module
+     */
+    ObservableList<Lesson> getModuleLessons(Code code);
+
+//    ObservableList<Task> getOverdueTasks(LocalDate date);
```

Comment:
Is this supposed to be commented out?

Reply:
Yes, because as we have discussed this will be a "bonus" feature. I just leave it there so that later on we remember